api/v1beta2/moduletemplate_types.go

```diff
@@ -126,6 +126,30 @@ type ModuleTemplateSpec struct {
 
 	// CustomStateCheck is deprecated.
 	CustomStateCheck []*CustomStateCheck `json:"customStateCheck,omitempty"`
+
+	// Info is the information about the module.
+	// +optional
+	Info ModuleInfo `json:"info,omitempty"`
+}
+
+type ModuleInfo struct {
+	// Repository is the link to the repository of the module.
+	Repository string `json:"repository"`
+
+	// Documentation is the link to the documentation of the module.
+	Documentation string `json:"documentation"`
+
+	// Icons is a list of icons associated with the module.
+	// +optional
+	Icons []ModuleIcon `json:"icons"`
```

Comment:
```suggestion
	// Icons is a list of icons associated with the module.
	// +optional
	// +listType=map
	// +listMapKey=name
	Icons []ModuleIcon `json:"icons"`
```

To force unique names.